odl/tomo/backends/astra_setup.py

```diff
@@ -0,0 +1,581 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Helper functions to prepare ASTRA algorithms.
+
+This module contains utility functions to convert data structures from the
+`odl.tomo.geometry` representation to ASTRA's data structures, including:
+
+* volume geometries
+* projection geometries
+* data arrays
+* algorithm configuration dictionaries
+* projectors and backprojectors
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from math import sin, cos
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+try:
+    import astra
+except ImportError:
+    pass
+import numpy as np
+
+# Internal
+from odl.space.ntuples import FnVector
+from odl.discr.grid import RegularGrid
+from odl.discr.lp_discr import DiscreteLp, DiscreteLpVector
+from odl.tomo.geometry.geometry import Geometry
```

Comment:
What's the guideline here? Should one import from the submodules or the top module?